---
title: "Data exploration"
author: "Selcan Aydin"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_depth: 4
    toc_float: 
      collapsed: false
      smooth_scroll: false
    df_print: paged
    code_folding: hide
    theme: flatly
---

# Setup and initial examples
 
<<<<<<< HEAD
First, let's call the `here` package to set our working directory. Make sure you opened the Rstudio project. This should tell us that we are at /path-to-git-repo/RNAseq_rotation_2021 . I am also loading the `tidyverse` and `ggpubr`packages. I also load all the functions I will use in a notebook here in the first code chunk as well. I included the function `create_dt` to make beatiful looking interactive tables that uses the `DT` package below. 

```{r load_packages_functions}

## Load packages and functions

#First, let's call the `here` package to set our working directory. Make sure you opened the Rstudio project. This should tell us that we are at /path-to-git-repo/RNAseq_rotation_2021 . I am also loading the `tidyverse` and `ggpubr`packages. I also load all the functions I will use in a notebook here in the first code chunk as well. I included the function `create_dt` to make beatiful looking interactive tables that uses the `DT` package below. 


### Packages
library(ggpubr)
library(tidyverse)
library(here) # you can install it if you don't have it: install.packages("here")


### Functions
# Making downloadable data tables
# https://www.r-bloggers.com/vignette-downloadable-tables-in-rmarkdown-with-the-dt-package/
create_dt <- function(x){
  DT::datatable(x,
                extensions = 'Buttons',
                rownames = FALSE, filter="top",
                options = list(dom = 'Blfrtip',
                               buttons = c('copy', 'csv', 'excel'),
                               pageLength = 10, scrollX=T))
}


```

You can record all the new packages installed using the `snapshot` function from `renv` package. 
```{r}
renv::settings$snapshot.type("simple")
```


## Load data

Let's now load in the data, it should be  located at /path-to-git-repo/RNAseq_rotation_2021/_data 

##loading paired end data
```{r}
mESC_paired_end<-  load("~/git/RNAseq_rotation_2021/_data/DO185_mESC_emase_m4_noProbs_correctedIDs_02012021.RData")
mESC_paired_end
```
```{r}
esc_expr_pe <- expr 

esc_raw_expr_pe <- raw.expr

esc_covar_pe <- covar 

rm(expr, covar, raw.expr, exprNotes, mixupNotes)

```


##loading r1 data
```{r}
mESC_r1<-load("~/git/RNAseq_rotation_2021/_data/DO185_mESC_emase_m4_noProbs_correctedIDs_02012021.RData")
mESC_r1
```

```{r}
esc_expr_r1 <- expr 

esc_raw_expr_r1 <- raw.expr

esc_covar_r1<- covar 

rm(expr, covar, raw.expr, exprNotes, mixupNotes)

```


 I will convert them to tibbles, then use `datatable` function from `DT` package to display the first couple lines as examples.

```{r}
esc_raw_expr_r1 %>% 
  as_tibble(rownames = "ensembl_gene_id") %>% 
 select (1:1000) %>%
  create_dt()
```

```{r}
expr %>%
  as_tibble (rownames = "ensembl_gene_id") -> mESC_paired_end_expr
```


Gene expression counts from only first set of reads (esc_raw_expr_r1) and the rownames are ensembl gene ids.

```{r}
esc_raw_expr_r1 %>% 
  as_tibble( rownames = "ensembl_gene_id") %>% # convert to tibble with the rownames as a new column labeled ensembl_gene_id
  select(1:6) %>%  # select first 5 columns
  slice( 1:10) %>%  # slice first 5 rows
  create_dt() # formatting
```


## Subsetting and plotting with data

First, let's convert the data tables into tibbles and save into new variables for easy access.

```{r make_tibbles}
esc_raw_expr_r1 %>% 
  as_tibble( rownames = "ensembl_gene_id") -> esc_raw_expr_r1_tibble # saving the tibble version to a new variable

esc_raw_expr_pe %>% 
  as_tibble( rownames = "ensembl_gene_id") -> esc_raw_expr_pe_tibble # saving the tibble version to a new variable

esc_covar %>% 
  as_tibble() -> esc_covar_tibble # saving the tibble version to a new variable

```


### Comparing counts of across samples: Scatterplot

```{r sample_correlation, fig.height=4, fig.width=6}

esc_raw_expr_r1_tibble %>% 
  ggplot() + # calling the plotting function
  aes( x = `ENSMUSG00000000001	`, 
       y = `PB357.07_repA`) + # now we can set the aesthetics, what goes on which axes, what color... you can use variable/column names inside here.
  geom_point( size = 2 ) + # now we call the type of plot we want to see. in this case we want points. and set related parameters.
  theme_pubclean() # theme: visual properties of the plot.

```

```{r}
esc_raw_expr_r1_tibble %>%slice(1:5) %>% 
  ggplot() + geom_bar(mapping = (aes(x =`PB357.07_repA`, fill =ensembl_gene_id)))
```


### Gene expression count distribution: Histogram

```{r histogram_example, fig.width=6, fig.height=4}

esc_raw_expr_r1_tibble  %>% 
  ggplot() + # call ggplot to make the plot!
  aes( x = `PB357.02_repA`) + # aesthetics determine which variable is being plotted and how it is being used. you can use column names here. 
  geom_histogram( bins = 1000, col = "blue", alpha = 0.5) + # which plot to show
  geom_histogram( aes( x = `PB357.07_repA`), col = "red", bins = 1000, alpha = 0.4)+
  theme_pubclean() +
  scale_x_log10()

```

### Gene expression count distribution: Density plot

```{r density_example}

esc_raw_expr_r1_tibble  %>% 
  ggplot() + # call ggplot to make the plot!
  aes( x = `PB357.02_repA`) + # aesthetics determine which variable is being plotted and how it is being used. you can use column names here. 
  geom_density(  col = "blue", alpha = 0.5, adjust = 0.5) + # which plot to show
  geom_density( aes( x = `PB357.07_repA`), col = "red",  alpha = 0.1, adjust =0.5)+
  theme_pubclean() +
  scale_x_log10()

```



### Comparing counts of genes: Scatterplot

We can transpose the data table and then convert to a tibble to get genes on columns and samples in rows. 

```{r}
esc_raw_expr_r1 %>% 
  t() %>% 
  as_tibble( rownames = "sample_id") -> esc_raw_expr_r1_tibble_t

esc_raw_expr_r1_tibble_t %>% 
  slice(1:6) %>% 
  select(1:10) %>% 
 create_dt()

```


```{r gene_correlation, fig.height=4, fig.width=6}

esc_raw_expr_r1_tibble_t %>% slice(1:10) %>% 
  ggplot() + # calling the plotting function
  aes( x = `ENSMUSG00000000056`, 
       y = `ENSMUSG00000000028`, color = sample_id) + # now we can set the aesthetics, what goes on which axes, what color... you can use variable/column names inside here.
  geom_point( size = 2 ) + # now we call the type of plot we want to see. in this case we want points. and set related parameters.
  theme_pubclean() # theme: visual properties of the plot.

```

```{r}
esc_raw_expr_r1_tibble_t %>% slice(1:2) %>% 
  ggplot() + # calling the plotting function
  aes( x = `ENSMUSG00000000056`, 
       y = `ENSMUSG00000000028`, shape = sample_id) + # now we can set the aesthetics, what goes on which axes, what color... you can use variable/column names inside here.
  geom_point( size = 2 ) + # now we call the type of plot we want to see. in this case we want points. and set related parameters.
  theme_pubclean() # theme: visual properties of the plot.
```


```{r}
esc_raw_expr_r1_tibble_t %>% slice(1:2) %>% 
  ggplot() + geom_bar(mapping = (aes(x =`ENSMUSG00000000028`, fill = sample_id)))
 
```

```{r}
esc_raw_expr_r1_tibble_t %>% slice(1:10) %>% 
  ggplot() + # calling the plotting function
  aes( x = `ENSMUSG00000000056`, 
       y = `ENSMUSG00000000028`, color = sample_id) + # now we can set the aesthetics, what goes on which axes, what color... you can use variable/column names inside here.
  geom_point() + # now we call the type of plot we want to see. in this case we want points. and set related parameters.
  theme_pubclean() # theme: visual properties of the plot.
```


working on pivot longer code
```{r}
esc_raw_expr_r1_tibble_t %>% slice(1:5) %>% select (1:10) %>% pivot_longer()

```


## Subset, reformat and plot

First, let's convert the data tables into tibbles and save into new variables for easy access.

```{r make_tibbles}
esc_raw_expr_r1 %>% 
  as_tibble( rownames = "ensembl_gene_id") -> esc_raw_expr_r1_tibble # saving the tibble version to a new variable

esc_raw_expr_pe %>% 
  as_tibble( rownames = "ensembl_gene_id") -> esc_raw_expr_pe_tibble # saving the tibble version to a new variable

esc_covar %>% 
  as_tibble() -> esc_covar_tibble # saving the tibble version to a new variable

```


### Comparing counts across samples: Scatterplot

```{r sample_correlation, fig.height=4, fig.width=6}

esc_raw_expr_r1_tibble %>% 
  ggplot() + # calling the plotting function
  aes( y = PB357.07_repA, 
       x = PB357.09_repA) + # now we can set the aesthetics, what goes on which axes, what color... you can use variable/column names inside here.
  geom_point( size = 2 ) + # now we call the type of plot we want to see. in this case we want points. and set related parameters.
  theme_pubclean() # theme: visual properties of the plot.

```

### Gene expression count distribution: Histogram

```{r histogram_example, fig.width=6, fig.height=4}

esc_raw_expr_r1_tibble  %>% 
  ggplot() + # call ggplot to make the plot!
  aes( x = `PB357.02_repA`) + # aesthetics determine which variable is being plotted and how it is being used. you can use column names here. 
  geom_histogram( bins = 1000, col = "blue", alpha = 0.5) + # which plot to show
  geom_histogram( aes( x = `PB357.07_repA`), col = "red", bins = 1000, alpha = 0.4)+
  theme_pubclean() +
  scale_x_log10()

```

### Gene expression count distribution: Density plot

```{r density_example}

esc_raw_expr_r1_tibble  %>% 
  ggplot() + # call ggplot to make the plot!
  aes( x = `PB357.02_repA`) + # aesthetics determine which variable is being plotted and how it is being used. you can use column names here. 
  geom_density(  col = "blue", alpha = 0.5, adjust = 0.5) + # which plot to show
  geom_density( aes( x = `PB357.07_repA`), col = "red",  alpha = 0.1, adjust =0.5)+
  theme_pubclean() +
  scale_x_log10()

```

### Comparing counts of genes: Scatterplot

We can transpose the data table and then convert to a tibble to get genes on columns and samples in rows. 

```{r}
esc_raw_expr_r1 %>% 
  t() %>% 
  as_tibble( rownames = "sample_id") -> esc_raw_expr_r1_tibble_t

esc_raw_expr_r1_tibble_t %>% 
  slice(1:6) %>% 
  select(1:10) %>% 
 create_dt()

```


```{r gene_correlation, fig.height=4, fig.width=6}

esc_raw_expr_r1_tibble_t %>% 
  ggplot() + # calling the plotting function
  aes( x = `ENSMUSG00000000056`, 
       y = `ENSMUSG00000000028`, 
       color = sample_id) + # now we can set the aesthetics, what goes on which axes, what color... you can use variable/column names inside here.
  geom_point( size = 2 , # now we call the type of plot we want to see. in this case we want points. and set related parameters.
              show.legend = FALSE) + 
  theme_pubclean() # theme: visual properties of the plot.

```

# Comparing data sets

Let's start comparing the two data sets. 

## Exercise 1: Plot mean gene expression distribution across each data set. 

```{r}
mESC_paired_end_expr %>% 
  t() %>% 
  as_tibble( rownames = "sample_id") -> mESC_paired_end_expr_t

```


```{r}
mESC_paired_end_expr_t %>%
  select (1:1000) %>%
  create_dt()
```

```{r}
colMeans(mESC_paired_end_expr_t)
```

```{r mean_gene_expression_histogram_pe, fig.width=8, fig.height=6}}

# add code here

```


## Exercise 2: Plot mean gene expression distribution of both datasets on the same plot and color differently.

```{r histogram_read1_pe}

# add code here

```

## Exercise 3: Identify most/least variable (n = 1000) genes and look at overlap.

## Exercise 4: Highlight common most/least variable genes in each data set.

## Exercise 5: Compare the mean gene expression of a single gene between the two datasets. 

Pick a few genes and compare transcript abundance across the two data sets. Do they have similar means or variation? What kind of plot would you make to visualize your results? Which statistics would you use to look at the agreement between the two data sets? 

## Exercise 6: Please make a scatterplot containing a regression line, pearson's R and p-value of the agreement looking at transcript abundance of a single gene across the two data sets. 

## Exercise 7: Add sample annotation (ex: sex of the samples) and re-make the plots with sex annotation. 

In which plots would it make a difference? Do you have to remake all of them? What else can you add to the sample annotation? 

# DESeq2

Here is the tutorial for DESEQ2: http://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html

## Exercise 8: Which expression matrix would we use in DESeq2? Expression matrix containing raw values or normalized/batch corrected values?

## Exercise 9: Can you convert the data tables containing transcript abundance and covarietes to the format that DESeq2 wants? What formula will you use to describe the covariates?

## Exercise 10: Subset the data tables to 5 male/female cell lines and convert to DESeq2 format.

## Exercise 11: Run DESeq2 with the PE read data set and compare Male/Female lines to each other.

## Exercise 12: Run DESeq2 with the both data sets and compare PE/single read counts to each other.




