---
title: "read_comparrisons"
author: "sherrea"
date: "2/19/2021"
output: html_document
---
This markdown will be used to compare expression data between paired end and single end reads

#Start by setting the working directory and loading the required packages

```{r}

library(here)
library(ggpubr)
library(tidyverse)
```

```{r}
renv::snapshot
```


##Load the paired end and single end data
```{r}
#loading pe data

mESC_paired_end <-  load( here("_data/DO185_mESC_paired_emase_m4_noProbs_correctedIDs_02012021.RData"))
mESC_paired_end

esc_expr_pe <- expr 

esc_raw_expr_pe <- raw.expr

esc_covar_pe <- covar 

esc_matches_pe <- matches

rm(expr, covar, raw.expr, exprNotes, mixupNotes, matches)


#loading r1 data

mESC_r1 <-load( here("_data/DO185_mESC_emase_m4_noProbs_correctedIDs_02012021.RData"))
mESC_r1

esc_expr_r1 <- expr 

esc_raw_expr_r1 <- raw.expr

esc_covar_r1 <- covar 

esc_matches_r1 <- matches

rm(expr, covar, raw.expr, exprNotes, mixupNotes, matches)

#loading r2 data

mESC_r2 <-load( here("_data/DO185_mESC_emase_m4_noProbs_correctedIDs_read2_02252021.RData"))
mESC_r2

esc_expr_r2 <- expr 

esc_raw_expr_r2 <- raw.expr

esc_covar_r2 <- covar 

esc_matches_r2 <- matches

rm(expr, covar, raw.expr, exprNotes, mixupNotes, matches)

```

##Right now the genes are in rows and the samples are in columns. I will convert the expression data to tibbles, then transposes the data. At each step a new variable will be created.
After this point I will have three variables for each data set (data frame, converted to tibbles, transposed data frame as a tibble).

```{r}
#convert to tibbles
esc_expr_pe %>% 
  as_tibble(rownames = "ensembl_gene_id", colnames = "sample_id") -> esc_expr_pe_tibble

esc_expr_r1 %>%
  as_tibble(rownames = "ensembl_gene_id", colnames = "sample_id") -> esc_expr_r1_tibble

esc_expr_r2 %>%
  as_tibble(rownames = "ensembl_gene_id", colnames = "sample_id") -> esc_expr_r2_tibble

#transpose tibbles
esc_expr_pe %>%
  t() %>%
  as_tibble( rownames = 'sample_id', colnames = "ensembl_gene_id") -> esc_expr_pe_tibble_t

 esc_expr_r1 %>%
  t() %>%
  as_tibble( rownames = 'sample_id', colnames = "ensembl_gene_id") -> esc_expr_r1_tibble_t
 
 esc_expr_r2 %>%
  t() %>%
  as_tibble( rownames = 'sample_id', colnames = "ensembl_gene_id") -> esc_expr_r2_tibble_t
```


I will now start comparing the different reads. First, I will look at the the expression of the most abundant and least abundant transcripts within each set to investigate the similarities or differences between them. I will be using the mean abundance of the transcripts.

I will calculate two different mean values. One using the first 1000 transcripts from each data set and on using the entire data set.
##Slice chooses rows and select chooses columns
```{r}
#Calculate the mean using the first 1000 rows of the data set
esc_expr_pe_tibble_t %>% select(1:1000) %>% 
  summarise_if( is.numeric, mean, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>% 
  rename( "mean" = "V1") -> esc_expr_pe_mean

esc_expr_r1_tibble_t  %>% select(1:1000) %>% 
  summarise_if( is.numeric, mean, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "mean" = "V1") -> esc_expr_r1_mean

esc_expr_r2_tibble_t  %>% select(1:1000) %>% 
  summarise_if( is.numeric, mean, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "mean" = "V1") -> esc_expr_r2_mean

#Arrange in ascending and descending order
esc_expr_pe_mean %>% 
  arrange(mean) -> least_abundant_esc_expr_pe

esc_expr_pe_mean %>%  
  arrange(-mean) -> most_abundant_esc_expr_pe

esc_expr_r1_mean %>%
  arrange(mean) -> least_abundant_esc_expr_r1

esc_expr_r1_mean %>%
  arrange(-mean) -> most_abundant_esc_expr_r1

esc_expr_r2_mean %>%
  arrange(mean) -> least_abundant_esc_expr_r2

esc_expr_r2_mean %>% 
  arrange(-mean) -> most_abundant_esc_expr_r2

#Calculate the mean using the entire data set
esc_expr_pe_tibble_t %>% 
  summarise_if( is.numeric, mean, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>% 
  rename( "mean" = "V1") -> esc_expr_pe_mean_t

esc_expr_r1_tibble_t  %>% 
  summarise_if( is.numeric, mean, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "mean" = "V1") -> esc_expr_r1_mean_t

esc_expr_r2_tibble_t  %>% 
  summarise_if( is.numeric, mean, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "mean" = "V1") -> esc_expr_r2_mean_t

#Arrange in ascending and descending order
esc_expr_pe_mean_t %>% 
  arrange(mean) -> least_abundant_esc_expr_pe_t

esc_expr_pe_mean_t %>%  
  arrange(-mean) -> most_abundant_esc_expr_pe_t

esc_expr_r1_mean_t %>%
  arrange(mean) -> least_abundant_esc_expr_r1_t

esc_expr_r1_mean_t %>%
  arrange(-mean) -> most_abundant_esc_expr_r1_t

esc_expr_r2_mean_t %>%
  arrange(mean) -> least_abundant_esc_expr_r2_t

esc_expr_r2_mean_t %>% 
  arrange(-mean) -> most_abundant_esc_expr_r2_t

most_abundant_esc_expr_pe_t
most_abundant_esc_expr_r1_t

```

##Use bar plots to visualise the 100 most and least abundant transcripts for each data set. Mean values calculated using the entire data set will be used.
```{r}

#plot most and lest abundant transcripts
least_abundant_esc_expr_pe_t %>% slice (1:100) %>%
  ggplot() + 
  ggtitle("Average values of the least abundant \npaired end read transcripts") + 
  aes( x = ensembl_gene_id,  y = mean)  +
   labs(x = "Ensembl gene ID") +
  geom_col(colour = "black", fill = "light blue") +
 theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

most_abundant_esc_expr_pe_t %>%  slice (1:100) %>%
  ggplot() + 
   ggtitle("Average values of the most abundant \npaired end read transcripts") +
  aes( x = ensembl_gene_id,  y = mean)  +
  labs(x = "Ensembl gene ID") +
  geom_col(colour = "black", fill = "light green") +
 theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

least_abundant_esc_expr_r1_t %>%  slice (1:100) %>%
  ggplot() + 
   ggtitle("Average values of the least abundant \nsigle end read 1 transcripts") +
  aes( x = ensembl_gene_id,  y = mean)  +
  labs(x = "Ensembl gene ID") +
  geom_col(colour = "black", fill = "cyan") +
   theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

most_abundant_esc_expr_r1_t %>%  slice (1:100) %>%
  ggplot() + 
   ggtitle("Average values of the most abundant \nsigle end read 1 transcripts") +
  aes( x = ensembl_gene_id,  y = mean)  +
  labs(x = "Ensembl gene ID") +
  geom_col(colour = "black", fill = "green") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

least_abundant_esc_expr_r2_t %>%  slice (1:100) %>%
  ggplot() + 
   ggtitle("Average values of the least abundant \nsigle end read 2 transcripts") +
  aes( x = ensembl_gene_id,  y = mean)  +
  labs(x = "Ensembl gene ID") +
  geom_col(colour = "black", fill ="royal blue 1") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())

most_abundant_esc_expr_r2_t %>%  slice (1:100) %>%
  ggplot() + 
   ggtitle("Average values of the most abundant \nsigle end read 2 transcripts") +
  aes( x = ensembl_gene_id,  y = mean)  +
  labs(x = "Ensembl gene ID") +
  geom_col(colour = "black", fill = "green4") +
 theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())




```

##Use a histogram to plot the averge transcript distribution for each read, using both sets of averages calculated.
```{r}
#plot distribution of transcript averages
esc_expr_pe_mean %>% 
  ggplot() + 
   ggtitle("Distribution of transcript averages \nfor paired end reads") +
  aes(x = mean) + 
  geom_histogram(bins = 50, colour = "black", fill = "gray 88") 
  

esc_expr_r1_mean %>% 
  ggplot() + 
    ggtitle("Distribution of transcript averages \nfor single end read 1") +
   labs(x = "transcript expression average") + 
  aes(x = mean) + 
  geom_histogram(bins = 50, colour = "black", fill = "coral")

esc_expr_r2_mean %>% 
  ggplot() + 
  ggtitle("Distribution of transcript averages \nfor single end read 2") +
   labs(x = "transcript expression average") + 
  aes(x = mean) + 
  geom_histogram(bins = 50, colour = "black", fill = "cadet blue 1")

esc_expr_pe_mean_t %>% 
  ggplot() + 
   ggtitle("Distribution of transcript averages \nfor paired end reads") +
  aes(x = mean) + 
  geom_histogram(bins = 100, colour = "black", fill = "gray 88") 
  

esc_expr_r1_mean_t %>% 
  ggplot() + 
    ggtitle("Distribution of transcript averages \nfor single end read 1") +
   labs(x = "transcript expression average") + 
  aes(x = mean) + 
  geom_histogram(bins = 100, colour = "black", fill = "coral")

esc_expr_r2_mean_t %>% 
  ggplot() + 
  ggtitle("Distribution of transcript averages \nfor single end read 2") +
   labs(x = "transcript expression average") + 
  aes(x = mean) + 
  geom_histogram(bins = 100, colour = "black", fill = "cadet blue 1")
#the x axis scale may need to be changed to allow for a more readable plot
#the x axis label should be changed. Figure out how
```

##Determine if there are any common genes within the most and least abundant transcripts in each read and create data tables to display these. Both setes of averages calculated will be used.
```{r}
#function to create data tables
create_dt <- function(x){
  DT::datatable(x,
                extensions = 'Buttons',
                rownames = FALSE, filter="top",
                options = list(dom = 'Blfrtip',
                               buttons = c('copy', 'csv', 'excel'),
                               pageLength = 10, scrollX=T))}

#Using just the first 1000 genes
#between all reads
most_abundant_transcripts <- intersect(most_abundant_esc_expr_pe[,1], most_abundant_esc_expr_r1[,1], most_abundant_esc_expr_r2[,1])
most_abundant_transcripts %>% create_dt()

#between r1 and r2
most_abundant_transcripts_r1_r2 <- intersect(most_abundant_esc_expr_r1[,1], most_abundant_esc_expr_r2[,1])
most_abundant_transcripts_r1_r2 %>% create_dt()

#between r1 and pe
most_abundant_transcripts_r1_pe <- intersect(most_abundant_esc_expr_pe[,1], most_abundant_esc_expr_r1[,1])
most_abundant_transcripts_r1_pe %>% create_dt()

#between r2 and pe
most_abundant_transcripts_r2_pe <- intersect(most_abundant_esc_expr_pe[,1],  most_abundant_esc_expr_r2[,1])
most_abundant_transcripts_r2_pe %>% create_dt()

#between all reads
least_abundant_transcripts <- intersect(least_abundant_esc_expr_pe[,1], least_abundant_esc_expr_r1[,1], least_abundant_esc_expr_r2[,1])
least_abundant_transcripts %>% create_dt()

#between r1 and r2
least_abundant_transcripts_r1_r2 <- intersect(least_abundant_esc_expr_r1[,1], least_abundant_esc_expr_r2[,1])
least_abundant_transcripts_r1_r2 %>% create_dt()

#between r1 and pe
least_abundant_transcripts_r1_pe <- intersect(least_abundant_esc_expr_pe[,1], least_abundant_esc_expr_r1[,1])
least_abundant_transcripts_r1_pe %>% create_dt()

#between r2 and pe
least_abundant_transcripts_r2_pe <- intersect(least_abundant_esc_expr_pe[,1],  least_abundant_esc_expr_r2[,1])
least_abundant_transcripts_r2_pe %>% create_dt()

##using the entire data set
#between all reads
most_abundant_transcripts_t <- intersect(most_abundant_esc_expr_pe_t[,1], most_abundant_esc_expr_r1_t[,1], most_abundant_esc_expr_r2_t[,1])
most_abundant_transcripts_t %>% create_dt()

#between r1 and r2
most_abundant_transcripts_r1_r2_t <- intersect(most_abundant_esc_expr_r1_t[,1], most_abundant_esc_expr_r2_t[,1])
most_abundant_transcripts_r1_r2_t %>% create_dt()

#between r1 and pe
most_abundant_transcripts_r1_pe_t <- intersect(most_abundant_esc_expr_pe_t[,1], most_abundant_esc_expr_r1_t[,1])
most_abundant_transcripts_r1_pe_t %>% create_dt()

#between r2 and pe
most_abundant_transcripts_r2_pe_t <- intersect(most_abundant_esc_expr_pe_t[,1],  most_abundant_esc_expr_r2_t[,1])
most_abundant_transcripts_r2_pe_t %>% create_dt()

#between all reads
least_abundant_transcripts_t <- intersect(least_abundant_esc_expr_pe_t[,1], least_abundant_esc_expr_r1_t[,1], least_abundant_esc_expr_r2_t[,1])
least_abundant_transcripts_t %>% create_dt()

#between r1 and r2
least_abundant_transcripts_r1_r2_t <- intersect(least_abundant_esc_expr_r1_t[,1], least_abundant_esc_expr_r2_t[,1])
least_abundant_transcripts_r1_r2_t %>% create_dt()

#between r1 and pe
least_abundant_transcripts_r1_pe_t <- intersect(least_abundant_esc_expr_pe_t[,1], least_abundant_esc_expr_r1_t[,1])
least_abundant_transcripts_r1_pe_t %>% create_dt()

#between r2 and pe
least_abundant_transcripts_r2_pe_t <- intersect(least_abundant_esc_expr_pe_t[,1],  least_abundant_esc_expr_r2_t[,1])
least_abundant_transcripts_r2_pe_t %>% create_dt()


```


##Calculate the standard deviation of the transcripts, using the entire data sets, and create data tables that show the commow genes amongst the most and least variable transcripts
```{r}
#Calculate the sd
esc_expr_pe_tibble_t %>% 
  summarise_if( is.numeric, sd, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "sd" = "V1") -> esc_expr_pe_st_dev

esc_expr_pe_st_dev %>%
  arrange(sd) -> most_var_esc_expr_pe

esc_expr_pe_st_dev %>%
  arrange(-sd) -> least_var_esc_expr_pe

esc_expr_r1_tibble_t %>% 
  summarise_if(is.numeric, sd, na.rm = TRUE) %>%
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "sd" = "V1") -> esc_expr_r1_st_dev


esc_expr_r1_st_dev %>%
  arrange(sd) -> most_var_esc_expr_r1

esc_expr_r1_st_dev %>%
  arrange(-sd) -> least_var_esc_expr_r1

esc_expr_r2_tibble_t %>% 
  summarise_if( is.numeric, sd, na.rm = TRUE) %>% 
  t() %>% 
  as_tibble(rownames = 'ensembl_gene_id') %>%
  rename( "sd" = "V1") -> esc_expr_r2_st_dev

esc_expr_r2_st_dev %>%
  arrange(sd) -> most_var_esc_expr_r2

esc_expr_r2_st_dev %>%
  arrange(-sd) -> least_var_esc_expr_r2

#common transcripts among the most and least variable within each read
most_var_transcripts <- intersect(most_var_esc_expr_pe[,1], most_var_esc_expr_r1[,1], most_var_esc_expr_r2[,1])
most_var_transcripts %>% create_dt()

least_var_transcripts <- intersect(least_var_esc_expr_pe[,1], least_var_esc_expr_r1[,1], least_var_esc_expr_r2[,1])
least_var_transcripts %>% create_dt()
```

I will analyse the expression of pluripotent associated genes 
##Tidy the data by using pivot longer and create box plots to compare the expression of three genes associated with ESC pluripotency
```{r}
esc_expr_pe_tibble %>%
  pivot_longer(! ensembl_gene_id, names_to = "sample_id", values_to = "expression") -> esc_expr_pe_tibble_longer
esc_expr_pe_tibble_longer$read <- 'pe'

esc_expr_r1_tibble %>%
  pivot_longer(! ensembl_gene_id, names_to = "sample_id", values_to = "expression") -> esc_expr_r1_tibble_longer
esc_expr_r1_tibble_longer$read <- 'r1'

esc_expr_r2_tibble %>%
  pivot_longer(! ensembl_gene_id, names_to = "sample_id", values_to = "expression") -> esc_expr_r2_tibble_longer
  esc_expr_r2_tibble_longer$read <- 'r2'
  
rbind(esc_expr_pe_tibble_longer, esc_expr_r1_tibble_longer, esc_expr_r2_tibble_longer) -> esc_read_type



#ENSMUSG00000074637 sox2 protein coding
#ENSMUSG00000012396 nanog protein coding
#ENSMUSG00000022346 myc protein coding


esc_read_type %>% filter(ensembl_gene_id == "ENSMUSG00000074637") %>%
  ggplot() + 
  aes( x =(ensembl_gene_id), y = expression, fill = read)  +
  geom_boxplot() +
  theme_pubclean()

```


#Analyse the biotypes of the transcripts by creating a plot as well as a data table that list the biotype of each transcript
```{r}

gene_bio_types <-  load( here("_data/all.genes.wbio.RData"))
gene_bio_types

bio_types <- all.genes.wbio
bio_types

bio_types %>%
  as_tibble() -> bio_types_t

bio_types_t %>% create_dt() -> bio_types_dt

bio_types_t %>% 
  count(gene_biotype) -> biotype_counts

biotype_counts %>% 
  filter(n>25) %>%
  arrange(n) %>%
  mutate(gene_biotype=factor(gene_biotype, levels=gene_biotype)) %>%
  ggplot() +
  aes( x = gene_biotype, y = n) + 
  labs(x = "biotype", y = "number of genes") +
  geom_col(colour = "black", fill = "lightcyan") +
  coord_flip()
```
